### DISCUSSION TITLE
Jinja template not working while in async

### DISCUSSION THREAD

**Discussion by anthony6401** [apache/superset#16501] [Q&A / Help]
Hi, I am trying to use jinja template in superset with asynchronous queries enabled. I am using superset version 1.2.0. It works after waiting for a while, then pressing the run again (I assume this is due to caching). I tried disabling the `GLOBAL_ASYNC_QUERIES` in the config file, and it works fine after disabling it. Is it possible to use jinja templating while maintaining the asynchronous queries, and not having to wait for a while for the first load to be stored in cache before running it again?

Below is the error message that I got.

superset_worker          | [2021-08-30 10:46:23,385: ERROR/ForkPoolWorker-15] Working outside of request context.
superset_worker          |
superset_worker          | This typically means that you attempted to use functionality that needed
superset_worker          | an active HTTP request.  Consult the documentation on testing for
superset_worker          | information about how to avoid this problem.
superset_worker          | Traceback (most recent call last):
superset_worker          |   File "/app/superset/common/query_context.py", line 361, in get_df_payload
superset_worker          |     query_result = self.get_query_result(query_obj)
superset_worker          |   File "/app/superset/common/query_context.py", line 114, in get_query_result
superset_worker          |     result = self.datasource.query(query_object.to_dict())
superset_worker          |   File "/app/superset/connectors/sqla/models.py", line 1422, in query
superset_worker          |     query_str_ext = self.get_query_str_extended(query_obj)
superset_worker          |   File "/app/superset/connectors/sqla/models.py", line 776, in get_query_str_extended
superset_worker          |     sqlaq = self.get_sqla_query(**query_obj)
superset_worker          |   File "/app/superset/connectors/sqla/models.py", line 1149, in get_sqla_query
superset_worker          |     tbl = self.get_from_clause(template_processor)
superset_worker          |   File "/app/superset/connectors/sqla/models.py", line 805, in get_from_clause
superset_worker          |     from_sql = self.get_rendered_sql(template_processor)
superset_worker          |   File "/app/superset/connectors/sqla/models.py", line 826, in get_rendered_sql
superset_worker          |     sql = template_processor.process_template(sql)
superset_worker          |   File "/app/superset/jinja_context.py", line 318, in process_template
superset_worker          |     return template.render(context)
superset_worker          |   File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 1090, in render
superset_worker          |     self.environment.handle_exception()
superset_worker          |   File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 832, in handle_exception
superset_worker          |     reraise(*rewrite_traceback_stack(source=source))
superset_worker          |   File "/usr/local/lib/python3.7/site-packages/jinja2/_compat.py", line 28, in reraise
superset_worker          |     raise value.with_traceback(tb)
superset_worker          |   File "<template>", line 3, in top-level template code
superset_worker          |   File "/usr/local/lib/python3.7/site-packages/jinja2/sandbox.py", line 462, in call
superset_worker          |     return __context.call(__obj, *args, **kwargs)
superset_worker          |   File "/app/superset/jinja_context.py", line 218, in safe_proxy
superset_worker          |     return_value = func(*args, **kwargs)
superset_worker          |   File "/app/superset/jinja_context.py", line 207, in url_param
superset_worker          |     if request.args.get(param):
superset_worker          |   File "/usr/local/lib/python3.7/site-packages/werkzeug/local.py", line 347, in __getattr__
superset_worker          |     return getattr(self._get_current_object(), name)
superset_worker          |   File "/usr/local/lib/python3.7/site-packages/werkzeug/local.py", line 306, in _get_current_object
superset_worker          |     return self.__local()
superset_worker          |   File "/usr/local/lib/python3.7/site-packages/flask/globals.py", line 38, in _lookup_req_object
superset_worker          |     raise RuntimeError(_request_ctx_err_msg)
superset_worker          | RuntimeError: Working outside of request context.
superset_worker          |
superset_worker          | This typically means that you attempted to use functionality that needed
superset_worker          | an active HTTP request.  Consult the documentation on testing for
superset_worker          | information about how to avoid this problem.

![image](https://user-images.githubusercontent.com/65406233/131331992-9aa010aa-4197-4421-ab86-7934f97bcd18.png)

Step to reproduce:
1. Install Superset (as per the instruction in https://superset.apache.org/docs/installation/installing-superset-using-docker-compose)
2. Set `ENABLE_TEMPLATE_PROCESSING` to `True` and configure Superset to run in async
3. Open any chart that is not cached.

Thank you and any help is appreciated!

**Comment by xza-m** [comment]
u need add config to FEATURE_FLAGS in superset_config.py

**Reply by xza-m** [reply]
like this 
FEATURE_FLAGS = {
        "ALERT_REPORTS": True,
        "ENABLE_TEMPLATE_PROCESSING": True
}
